### PR TITLE
Notify releases

### PR DESCRIPTION
**Pull request by yacinehmito**
At Spendesk we use GitHub to track the releases of our product, then copy, reformat and paste the GitHub release notes to publish them on a dedicated Slack channel. This process is quite tedious.

This Pull Request introduces `release` to the list of listened events. The implementation is quite rudimentary as it simply posts the content of the release note.

This is WIP; manual tests have not been performed. I am waiting to see if you welcome this new feature and if there is anything else to be done (for example I didn't handle any unfurling and you may want that).

Cheers!

### CODE REVIEW

File: lib/messages/release.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    this.release = release;
+  }
+
+  getRenderedMessage() {
+    const { release } = this;
+    const { author } = release;
+
+    const message = {
+      ...super.getBaseMessage(),
+      title: `Release - ${release.name}`,
```
Comment (bkeepers) [inline lib/messages/release.js]: @wilhelmklopp I know we talked about ways of standardizing the "type" in messages, but I don't remember what we decided.

A few examples of what we do elsewhere:

- Comments: `Comment on #${this.issue.number} ${this.issue.title}`
- Issue/PR: `#${this.issue.number} ${this.issue.title}`
- Review: `Review on #${this.pull.number} ${this.pull.title}`
- Public: `${this.repository.full_name} is now public!`

Comment (bkeepers) [reply lib/messages/release.js]: This message will also need a `fallback`, which can be the same as `title`, but prefixed with `[owner/repo]`.

File: lib/activity/releases.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+const { Release } = require('../messages/release');
+
+async function releaseEvent(context, subscription, slack) {
+  const eventType = `${context.event}.${context.payload.action}`;
+
+  if (eventType === 'release.published') {
```
[the remark this answers is not in the document]
Comment (yacinehmito) [reply lib/activity/releases.js]: This doesn't seem to be possible. Indeed:
- This is not documented
- I tried editing a release and this does not send any event to the webhook
- I found [a post hat says so](https://platform.github.community/t/release-webhook-not-emitted-on-update/3772/5). It is relatively old but there are no signs of any changes.

**Comment by yacinehmito** [comment]
I have updated the branch with the requested changes.
Updating the message when the release note is edited does not seem to be possible though.

Also, by adding automatic subscriptions to the `release` events, what happens with channel that have already been set up? Will they start receiving releases or must they subscribe manually?

**Comment by bkeepers** [comment]
Thanks @yacinehmito, this is in production now!

**Comment by bethesque** [comment]
Awesome! Thanks so much @yacinehmito

**Comment by yacinehmito** [comment]
It was a pleasure! 😄

**Comment by aaron-sf** [comment]
I think the docs still need an update to reflect this? https://slack.com/intl/en-gb/help/articles/232289568-GitHub-for-Slack#decide-which-app-to-use

**Comment by chhh** [comment]
it's end of March 2020, two years down the road docs still don't say anything about releases notification

**Comment by ajacksified** [comment]
I too found this PR because the docs don't mention releases.

**Pull request merged**